packages/component-library/src/components/table-and-list/mt-data-table/mt-data-table.vue

```diff
@@ -453,23 +463,27 @@
                   v-if="!(disableSettingsTable && disableEdit && disableDelete)"
                   class="mt-data-table__table-context-button"
                 >
-                  <a v-if="!disableEdit" href="#" @click.prevent="$emit('open-details', data)">
-                    {{ $t("mt-data-table.contextButtons.edit") }}
-                  </a>
-                  <mt-context-button v-if="!(disableDelete && disableEdit)">
-                    <mt-context-menu-item
-                      v-if="!disableEdit"
-                      :label="$t('mt-data-table.contextButtons.edit')"
-                      @click="$emit('open-details', data)"
-                    />
-
-                    <mt-context-menu-item
-                      v-if="!disableDelete"
-                      type="critical"
-                      :label="$t('mt-data-table.contextButtons.delete')"
-                      @click="$emit('item-delete', data)"
-                    />
-                  </mt-context-button>
+                  <slot name="action-button" v-bind="{ data }">
```

Comment:
Same here. If someone uses this slot some logic with "disableEdit", etc. can get broken by using the slot wrong (e.g. the v-if before does not render the button even if the logic would make sense for it).

I would prefer to implement it in a more safe way with a slot just around the text. Then it is also customizable but is more stable.

```html
<a v-if="!disableEdit" href="#" @click.prevent="$emit('open-details', data)">
  <slot name="action-button" v-bind="{ data }">
    {{ $t("mt-data-table.contextButtons.edit") }}
  </slot>
</a>
```

Reply:
by this way, we can not change the action for this button, for example, I want to replace by a "view details" button instead of the edit button.